src/Rules/Violation.php

```diff
@@ -10,23 +10,26 @@ class Violation implements \JsonSerializable
 
     private ?int $line;
 
+    private ?string $filePath;
+
     private string $error;
 
-    public function __construct(string $fqcn, string $error, ?int $line = null)
+    public function __construct(string $fqcn, string $error, ?int $line = null, ?string $filePath = null)
```

Comment:
maybe we could make this construct private?

Reply:
doable, will tackle in a separate PR 🙇